meta.go

```diff
@@ -236,13 +236,23 @@ func setMetaValueRecursive(key string, value string, previousMeta interface{}) (
 			}
 		} else if string([]rune{char}) == "." {
 			// Value is object
+			keyHead := key[0:current]   // e.g. aaa.bbb -> aaa
 			childKey := key[current+1:] // e.g. aaa.bbb -> bbb
-			key = key[0:current]        // e.g. aaa.bbb -> aaa
 			obj := make(map[string]interface{})
+			var tmpValue interface{}
 			previousMetaMap := convertInterfaceToMap(previousMeta)
-			childKey, tmpValue := setMetaValueRecursive(childKey, value, previousMetaMap[key])
+			if previousMetaMap[keyHead] == nil {
+				childKey, tmpValue = setMetaValueRecursive(childKey, value, previousMetaMap)
+			} else {
+				// copy previous object only if it is map
+				previousObj := convertInterfaceToMap(previousMetaMap[keyHead])
+				if len(previousObj) != 0 {
```

Comment:
When `previousMetaMap[keyHead]` is just value (not map), `convertInterfaceToMap(previousMetaMap[keyHead])` returns nil.  nil map causes below panic: 
```
panic: assignment to entry in nil map [recovered]
	panic: assignment to entry in nil map
```
To avoid this, I added this check.